### PR TITLE
Improve stats table alignment

### PR DESCRIPTION
**Pull request by Janpot**
"⚡" occupies two characters but is counted as one (or is this just on my terminal?)

before:
Screenshot 2019-07-28 at 17 48 24

after:
Screenshot 2019-07-28 at 17 48 02

### CODE REVIEW

**Comment by timneutkens** [comment]
I haven't seen it appear that way for me.

**Comment by Janpot** [comment]
Which terminal do you use? It's probably just a problem with bash then.

**Comment by timneutkens** [comment]
Hyper + zsh

**Comment by huv1k** [comment]
I have it a little bit off too, but I think it's going to be off in other OS and terminals then.

**Comment by Janpot** [comment]
For what it's worth [`string-width`](https://www.npmjs.com/package/string-width) seems to report a width of 2 for '⚡'. Updated the logic to use that package for now.

File: packages/next/build/utils.ts [hunk 1]
```diff
@@ -71,16 +67,17 @@ export function printTreeView(
...
       const pageInfo = pageInfos.get(item)
 
+      const renderTarget = item.startsWith('/_')
+        ? ' '
+        : pageInfo && pageInfo.static
+        ? chalk.bold('⚡')
+        : serverless
+        ? 'λ'
+        : 'σ'
+      const maxWidth = Math.max(...[' ', 'λ', 'σ', '⚡'].map(stringWidth))
```
Comment (timneutkens) [inline packages/next/build/utils.ts]: Since these are static values, can't we hardcode the width?

Comment (Janpot) [reply packages/next/build/utils.ts]: No, `string-width` is purely a function of the string input as far as I can tell from digging through the code.

**Review by Timer — CHANGES_REQUESTED** [review]
Wouldn't this be solved by using `string-with` only? I don't think we need to add the space padding to the symbols.

File: packages/next/build/utils.ts [hunk 2]
```diff
@@ -70,12 +70,13 @@ export function printTreeView(
       messages.push([
         `${symbol} ${
           item.startsWith('/_')
-            ? ' '
+            ? '  '
```
Comment (Timer) [inline packages/next/build/utils.ts]: Meaning undo this.

File: packages/next/build/utils.ts [hunk 3]
```diff
@@ -70,12 +70,13 @@ export function printTreeView(
...
             : pageInfo && pageInfo.static
-            ? chalk.bold('⚡')
+            ? // '⚡' has a width of two characters in the terminal
+              chalk.bold('⚡')
             : serverless
-            ? 'λ'
-            : 'σ'
+            ? 'λ '
```
Comment (Timer) [inline packages/next/build/utils.ts]: And this.

File: packages/next/build/utils.ts [hunk 4]
```diff
@@ -70,12 +70,13 @@ export function printTreeView(
...
+            : 'σ '
```
Comment (Timer) [inline packages/next/build/utils.ts]: And this.

File: packages/next/build/utils.ts [hunk 5]
```diff
@@ -70,12 +70,13 @@ export function printTreeView(
       messages.push([
         `${symbol} ${
           item.startsWith('/_')
-            ? ' '
+            ? '  '
             : pageInfo && pageInfo.static
-            ? chalk.bold('⚡')
+            ? // '⚡' has a width of two characters in the terminal
```
Comment (Timer) [inline packages/next/build/utils.ts]: Comment probably isn't needed now since the package handles this nuanced detail.

**Review by Timer — CHANGES_REQUESTED** [review]

**Comment by Janpot** [comment]
@Timer contrary to what it looks like, these 3 first columns (tree symbol, render target, filepath) are all in one column of `textTable`. this is done so that they can have a header spanning those three columns (`Page`). `textTable` doesn't support headers spanning multiple columns. this has the consequence that we need to do the alignement manually, hence the space after every symbol except the ⚡. I added that comment to explain that we don't need a space after ⚡. To illustrate what it would look like with your suggestion:
Screenshot 2019-07-29 at 20 29 14
I can do that ofcourse if you prefer it, but IMO it looks better when the filepaths line up.

**Comment by Janpot** [comment]
@Timer extracted out another column in the front, hope this works for you. It looks like this now:
Screenshot 2019-07-29 at 23 30 50
Looks better imo

**Comment by ijjk** [comment]
## Stats from current PR

<details>
<summary>Click to expand stats</summary>

| | zeit/next.js canary | Janpot/next.js fix-outpu-alignment | Change |
| - | - | - | - |
| Build Duration | 23.6s | 23.3s | -307ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | ⚠️  +68 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.35 kB | 2.35 kB | ✓ |
| Client `_app` gzip Size | 1.07 kB | 1.07 kB | ✓ |
| Client `_app` Modern Size | 1.79 kB | 1.79 kB | ✓ |
| Client `_app` gzip Modern Size | 880 B | 880 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | ✓ |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 301 B | ✓ |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Base Rendered Size | 2.76 kB | 2.76 kB | ✓ |
| Build Dir Size | 1.44 MB | 1.44 MB | ✓ |

</details>


<details>
<summary>Click to expand serverless stats</summary>

| | zeit/next.js canary | Janpot/next.js fix-outpu-alignment | Change |
| - | - | - | - |
| Build Duration | 25.5s | 25.2s | -306ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | ⚠️  +68 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.35 kB | 2.35 kB | ✓ |
| Client `_app` gzip Size | 1.07 kB | 1.07 kB | ✓ |
| Client `_app` Modern Size | 1.79 kB | 1.79 kB | ✓ |
| Client `_app` gzip Modern Size | 880 B | 880 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | -1 B |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 300 B | -1 B |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Serverless `pages/link` Size | 256 kB | 256 kB | ✓ |
| Serverless `pages/link` gzip Size | 68.8 kB | 68.8 kB | ⚠️  +1 B |
| Serverless `pages/index` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/index` gzip Size | 66.6 kB | 66.6 kB | ✓ |
| Serverless `pages/_error` Size | 248 kB | 248 kB | ✓ |
| Serverless `pages/_error` gzip Size | 66.3 kB | 66.3 kB | ⚠️  +1 B |
| Serverless `pages/routerDirect` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/routerDirect` gzip Size | 66.5 kB | 66.5 kB | ⚠️  +1 B |
| Serverless `pages/withRouter` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/withRouter` gzip Size | 66.6 kB | 66.6 kB | ✓ |
| Build Dir Size | 2.64 MB | 2.64 MB | ✓ |

</details>

**Review by Timer — CHANGES_REQUESTED** [review]
This doesn't look correct on macOS:
image

**Comment by Timer** [comment]
Maybe we can try the `string-length` package?  I'm not sure if we can make all CLIs happy here, maybe we can use a non-special character?

**Comment by Janpot** [comment]
@Timer 
> This doesn't look correct on macOS:

I'm also on macOS, which terminal are you using?

> Maybe we can try the `string-length` package?

It just seems that some terminals display "⚡" as a halfwidth character and others as  a fullwidth character. So `string-length` wouldn't solve the problem neither.

> maybe we can use a non-special character?

That's probably the easiest solution, do you have a suggestion? There is another lightning bolt `☇`, but it doesn't look great.

Screenshot 2019-08-12 at 08 02 01

Maybe another greek letter?
Screenshot 2019-08-12 at 08 04 35

Or some sort of color coding

Screenshot 2019-08-12 at 08 07 24

**Comment by Timer** [comment]
I'm using the VSCode terminal for the above text, in Bash.

**Comment by Janpot** [comment]
Maybe related to https://github.com/Microsoft/vscode/issues/32840?

**Comment by Timer** [comment]
Fixed by https://github.com/zeit/next.js/pull/9281

**Pull request closed without merging**